Alpaca.Markets/RateThrottler.cs

```diff
@@ -79,16 +99,26 @@ internal sealed class RateThrottler : IThrottler, IDisposable
         /// <inheritdoc />
         public void Dispose()
         {
-            _semaphore.Dispose();
+            _throttleSemaphore.Dispose();
             _exitTimer.Dispose();
         }
 
         /// <inheritdoc />
         public void WaitToProceed()
         {
 
+            // Block until any all stops are over
+            while (_allStop && unchecked(_allStopUntil - Environment.TickCount) > 0)
```

Comment:
Performance issue - do not calculate this delay value twice.

Reply:
Added variable to hold the calculated value.  As I'm typing this reply, it occurs to me that the reason this while loop existed was with the expectation that if while this request is waiting, another request receives a 429 or 503, that it would extend the AllStop, causing everybody to wait until we stopped getting those statuses.  This logic depended upon consistency in the wait time, i.e. the 3000 ms.  Using randomness in the delay times means this request might be waiting 5000ms, and the next one truncates the wait to 1000ms.  It causes no harm, really, but makes the while loop ineffective.  Waiting arbitrary times is highly sub-optimal in any case, so it probably doesn't matter.